### PR TITLE
Adding deque to valid field types

### LINKED ISSUE #1934: Accept deque as sequence-like input data type
### Checks

* [x] I added a descriptive title to this issue
* [x] I have searched (google, github) for similar issues and couldn't find anything
* [x] I have read and followed [the docs](https://pydantic-docs.helpmanual.io/) and still think this feature/change is needed
* [x] After submitting this, I commit to one of:
    * Look through open issues and helped at least one other person
    * Hit the "watch" button on this repo to receive notifications and I commit to help at least 2 people that ask questions in the future
    * Implement a Pull Request for a confirmed bug

# Feature Request

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
             pydantic version: 1.6.1
            pydantic compiled: True
                 install path: /home/quan/.local/lib/python3.8/site-packages/pydantic
               python version: 3.8.2 (default, Jul 16 2020, 14:00:26)  [GCC 9.3.0]
                     platform: Linux-5.4.0-47-generic-x86_64-with-glibc2.29
     optional deps. installed: []

```

Code snippet:

```py
from collections import deque
from typing import List
from pydantic import BaseModel


class Ticket(BaseModel):
    numbers: List[int] = []


Ticket.parse_obj({'numbers': deque((1, 3, 5))})
```

Got error:

```py
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pydantic/main.py", line 455, in pydantic.main.BaseModel.parse_obj
  File "pydantic/main.py", line 346, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for Ticket
numbers
  value is not a valid list (type=type_error.list)

1 validation error for Ticket
numbers
  value is not a valid list (type=type_error.list)

```

I expect to get:

```py
Ticket(numbers=[1, 3, 5])
```

### PR DESCRIPTION
**Pull request by wozniakty**
## Change Summary

Got bored so I PR'd an issue from @hongquan:
Adding deque as a valid sequence type for fields

## Related issue number

Closes #1934 

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

**Review by hongquan — APPROVED** [review]
Look good to me

File: docs/usage/types.md [hunk 1]
```diff
@@ -90,6 +94,9 @@ with custom properties and validation.
 `typing.FrozenSet`
 : see [Typing Iterables](#typing-iterables) below for more detail on parsing and validation
 
+`typing.Deque`
+: see [Typing Iterables](#typing-iterables) below for more detail on parsing and validation
```
Comment (samuelcolvin) [inline docs/usage/types.md:98]: you mention "below" but I think you haven't added a deque case to example where these other tupes are demonstrated?

**Review by samuelcolvin — COMMENTED** [review]
One tiny thing, otherwise this looks great.

**Comment by samuelcolvin** [comment]
you'll also need to resolve conflicts.

File: docs/usage/types.md (hunk 1 above)
Comment (wozniakty) [reply docs/usage/types.md:98]: Woops, fixed!

**Comment by wozniakty** [comment]
@samuelcolvin comments addressed :+1:

**Comment by wozniakty** [comment]
@samuelcolvin updated, thanks for the input :+1: this PR gave me a _lot_ more context on how pydantic works, was fun to work on

File: docs/usage/types.md [hunk 2]
```diff
@@ -31,25 +31,29 @@ with custom properties and validation.
...
+: allows `list`, `tuple`, `set`, `frozenset`, `deque`, or generators and casts to a set;
   see `typing.Set` below for sub-type constraints
 
 `frozenset`
-: allows `list`, `tuple`, `set`, `frozenset`, or generators and casts to a frozen set;
+: allows `list`, `tuple`, `set`, `frozenset`, `deque`, or generators and casts to a frozen set;
   see `typing.FrozenSet` below for sub-type constraints
 
+`deque`
+: allows `list`, `tuple`, `set`, `frozenset`, `deque`, or generators and casts to a frozen set;
```
Comment (PrettyWood) [inline docs/usage/types.md:54]:
```suggestion
: allows `list`, `tuple`, `set`, `frozenset`, `deque`, or generators and casts to a deque;
```

**Review by PrettyWood — CHANGES_REQUESTED** [review]

File: docs/usage/types.md (hunk 2 above)
Comment (wozniakty) [reply docs/usage/types.md]: Thanks for catching that

**Review by PrettyWood — APPROVED** [review]
Thanks. LGTM!

**Pull request merged**